src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -267,7 +267,11 @@ def get_clouds(cli_ctx):
     # Start off with known clouds and apply config file on top of current config
     for c in KNOWN_CLOUDS:
         _config_add_cloud(config, c)
-    config.read(CLOUD_CONFIG_FILE)
+    try:
+        config.read(CLOUD_CONFIG_FILE)
+    except configparser.MissingSectionHeaderError:
+        # clouds.config file is in bad format. Reomve it.
+        os.remove(CLOUD_CONFIG_FILE)
```

Comment:
Can we verify (at least manually) that once the corrupt Cloud Config is removed that it is regenerated? We should also consider a debug log here that cloud config was corrupt and is being removed.

Reply:
Yes, it will regenerated with az account set command.
Debug log is added.